src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -208,21 +221,21 @@ pub trait MetadataChainExt: MetadataChain {
 
             for (decision, visitor) in decisions.iter_mut().zip(visitors.iter_mut()) {
                 match decision {
-                    Decision::Stop => {
+                    MetadataVisitorDecision::Stop => {
                         stopped_visitors += 1;
                     }
-                    Decision::NextOfType(type_flags) if type_flags.is_empty() => {
+                    MetadataVisitorDecision::NextOfType(type_flags) if type_flags.is_empty() => {
                         stopped_visitors += 1;
                     }
-                    Decision::Next => {
+                    MetadataVisitorDecision::Next => {
                         *decision = visitor.visit(hashed_block_ref)?;
                     }
-                    Decision::NextWithHash(requested_hash) => {
+                    MetadataVisitorDecision::NextWithHash(requested_hash) => {
                         if hash == *requested_hash {
                             *decision = visitor.visit(hashed_block_ref)?;
                         }
                     }
```

Comment:
> So, we are quickly finishing if any of the visitors return an error. I wonder if there would be a need to collect multiple `Result` of visitors. Something like displaying multiple validation errors in 1 run. (c) https://github.com/kamu-data/kamu-cli/pull/523#discussion_r1517482067

Reply:
I don't think we need it at the moment. We may need it later -- for example, in case of API processing. It's not hard to implement, but I think we can postpone it for now.

@zaychenko-sergei , what do you think?